### ISSUE TITLE
SHIP is missing block field on live blocks

### ISSUE THREAD

**Issue by fschoell** [eosnetworkfoundation/mandel#440]
Just tested on our Jungle4 SHIP node running the current mandel. Whenever we catch up to the live blocks the `Block` field starts to become null (until then the blocks are returned as expected). 

Request parameters, `FetchBlock` is set to true: 

```json
{
  "StartBlockNum": 20162274,
  "EndBlockNum": 4294967295,
  "MaxMessagesInFlight": 4294967295,
  "HavePositions": null,
  "IrreversibleOnly": false,
  "FetchBlock": true,
  "FetchTraces": true,
  "FetchDeltas": false
}
```

After catching up to the head block the `Block` field is missing, example:

```json
{
  "block": {
    "Head": {
      "BlockNum": 20162471,
      "BlockID": "0133a7a758c9fd333e8135b0922f34e00679fa6ba9bc18a843afe428c6365981"
    },
    "LastIrreversible": {
      "BlockNum": 20162138,
      "BlockID": "0133a65aeefd9d6f3c4d546fd7664edb062b2062b2bc4ce55ec5b9faf6ac76ce"
    },
    "ThisBlock": {
      "BlockNum": 20162471,
      "BlockID": "0133a7a758c9fd333e8135b0922f34e00679fa6ba9bc18a843afe428c6365981"
    },
    "PrevBlock": {
      "BlockNum": 20162470,
      "BlockID": "0133a7a61e39c23fcf84aa45883c46242c8089ce0232764ecdf0df9d732e0960"
    },
    "Block": null,
    "Traces": {
      "Elem": [
        {
          "TypeID": 0,
          "Impl": {
            "id": "3f7ff49ab9bd51a727247c1e2884897bc66baa4a2b431afdadc0002097b58869",
            "Status": "executed",
            "cpu_usage_us": 1,
            "net_usage_words": 0,
            "elapsed": 0,
            "net_usage": 0,
            "scheduled": false,
            "action_traces": [
              null
            ],
            "account_delta": null,
            "except": "",
            "error_code": 0,
            "failed_dtrx_trace": null,
            "partial": {
              "TypeID": 0,
              "Impl": {
                "Expiration": 0,
                "RefBlockNum": 0,
                "RefBlockPrefix": 0,
                "MaxNetUsageWords": 0,
                "MaxCpuUsageMs": 0,
                "DelaySec": 0,
                "TransactionExtensions": [],
                "Signatures": [],
                "ContextFreeData": ""
              }
            }
          }
        },
        {
          "TypeID": 0,
          "Impl": {
            "id": "54163324f8e748c388ef8a1b8f9160a1c3b39e37be5c9bc8b00a53492ab33c56",
            "Status": "executed",
            "cpu_usage_us": 221,
            "net_usage_words": 15,
            "elapsed": 0,
            "net_usage": 120,
            "scheduled": false,
            "action_traces": [
              null,
              {
                "TypeID": 1,
                "Impl": {
                  "ActionOrdinal": 2,
                  "CreatorActionOrdinal": 0,
                  "Receipt": {
                    "TypeID": 0,
                    "Impl": {
                      "Receiver": "eosmechanics",
                      "ActDigest": "270d4d212c5dc277ab6b2f83e545ff00f762ee6573070e07a2dda288bdc896d4",
                      "GlobalSequence": 21268528,
                      "RecvSequence": 545145,
                      "AuthSequence": [
                        {
                          "Account": "eosmechanics",
                          "Sequence": 545151
                        }
                      ],
                      "CodeSequence": 1,
                      "ABISequence": 1
                    }
                  },
                  "Receiver": "eosmechanics",
                  "Act": {
                    "Account": "eosmechanics",
                    "Name": "cpu",
                    "Authorization": [
                      {
                        "actor": "eosmechanics",
                        "permission": "benchmark"
                      }
                    ],
                    "Data": "e30="
                  },
                  "ContextFree": false,
                  "Elapsed": 0,
                  "Console": "",
                  "AccountRamDeltas": [],
                  "Except": "",
                  "ErrorCode": 0,
                  "ReturnValue": ""
                }
              }
            ],
            "account_delta": null,
            "except": "",
            "error_code": 0,
            "failed_dtrx_trace": null,
            "partial": {
              "TypeID": 0,
              "Impl": {
                "Expiration": 1655725569,
                "RefBlockNum": 42586,
                "RefBlockPrefix": 1867795772,
                "MaxNetUsageWords": 0,
                "MaxCpuUsageMs": 0,
                "DelaySec": 0,
                "TransactionExtensions": [],
                "Signatures": [
                  "SIG_K1_K1C731F97cowJNpEmsYPwpyvvw5doMw46bNCMsyUKYaoF6jPfTn4KLy3XVqDLNY6k2HrU83J8spBf5E9kcU3TFTPXji5FT"
                ],
                "ContextFreeData": ""
              }
            }
          }
        }
      ]
    },
    "Deltas": null
  }
}
```

Older block ranges return fine and requesting the above block later again will also return it as expected:

```json
{
  "block": {
    "Head": {
      "BlockNum": 20165806,
      "BlockID": "0133b4aeb6ea4118bddadf198228f4fe26df974ee2651029a86fda44426abf33"
    },
    "LastIrreversible": {
      "BlockNum": 20165474,
      "BlockID": "0133b362616ebdbf47cc842d50b0773a7ad575f42e9da82716894e19ef0d0036"
    },
    "ThisBlock": {
      "BlockNum": 20162471,
      "BlockID": "0133a7a758c9fd333e8135b0922f34e00679fa6ba9bc18a843afe428c6365981"
    },
    "PrevBlock": {
      "BlockNum": 20162470,
      "BlockID": "0133a7a61e39c23fcf84aa45883c46242c8089ce0232764ecdf0df9d732e0960"
    },
    "Block": {
      "timestamp": "2022-06-20T13:45:40",
      "producer": "eosnationftw",
      "confirmed": 0,
      "previous": "0133a7a61e39c23fcf84aa45883c46242c8089ce0232764ecdf0df9d732e0960",
      "transaction_mroot": "09c648d53ea8efeb8da1fcfd8dc1fbbdbf2bc3a7895b9dc1016fc2271f9faebe",
      "action_mroot": "adee57ce1c4b7aa9301c708152a6e2f49db5265791dfb25d82d341347ff9dc8c",
      "schedule_version": 25,
      "header_extensions": [],
      "ProducerSignature": "SIG_K1_K7HLw3my1iVNVKqREJVPGm8b7xrSAxciDNMwayxm2EQtfasJnNrq71GvZYmapyKN2aqHJx8NbS4HqjrczemCTi6guTX6gw",
      "Transactions": [
        {
          "status": "executed",
          "cpu_usage_us": 221,
          "net_usage_words": 15,
          "Trx": {
            "TypeID": 1,
            "Impl": {
              "signatures": [
                "SIG_K1_K1C731F97cowJNpEmsYPwpyvvw5doMw46bNCMsyUKYaoF6jPfTn4KLy3XVqDLNY6k2HrU83J8spBf5E9kcU3TFTPXji5FT"
              ],
              "compression": "none",
              "packed_context_free_data": "",
              "packed_trx": "015eb0625aa63c4d546f0000000100408c7a02ea3055000000000085269d0008416c9fa4dfe105000180919ba62125315500000000000074450180919ba621253155000080d7c886a63a0000"
            }
          }
        }
      ],
      "BlockExtensions": []
    },
    "Traces": {
      "Elem": [
        {
          "TypeID": 0,
          "Impl": {
            "id": "3f7ff49ab9bd51a727247c1e2884897bc66baa4a2b431afdadc0002097b58869",
            "Status": "executed",
            "cpu_usage_us": 1,
            "net_usage_words": 0,
            "elapsed": 0,
            "net_usage": 0,
            "scheduled": false,
            "action_traces": [
              null
            ],
            "account_delta": null,
            "except": "",
            "error_code": 0,
            "failed_dtrx_trace": null,
            "partial": {
              "TypeID": 0,
              "Impl": {
                "Expiration": 0,
                "RefBlockNum": 0,
                "RefBlockPrefix": 0,
                "MaxNetUsageWords": 0,
                "MaxCpuUsageMs": 0,
                "DelaySec": 0,
                "TransactionExtensions": [],
                "Signatures": [],
                "ContextFreeData": ""
              }
            }
          }
        },
        {
          "TypeID": 0,
          "Impl": {
            "id": "54163324f8e748c388ef8a1b8f9160a1c3b39e37be5c9bc8b00a53492ab33c56",
            "Status": "executed",
            "cpu_usage_us": 221,
            "net_usage_words": 15,
            "elapsed": 0,
            "net_usage": 120,
            "scheduled": false,
            "action_traces": [
              null,
              {
                "TypeID": 1,
                "Impl": {
                  "ActionOrdinal": 2,
                  "CreatorActionOrdinal": 0,
                  "Receipt": {
                    "TypeID": 0,
                    "Impl": {
                      "Receiver": "eosmechanics",
                      "ActDigest": "270d4d212c5dc277ab6b2f83e545ff00f762ee6573070e07a2dda288bdc896d4",
                      "GlobalSequence": 21268528,
                      "RecvSequence": 545145,
                      "AuthSequence": [
                        {
                          "Account": "eosmechanics",
                          "Sequence": 545151
                        }
                      ],
                      "CodeSequence": 1,
                      "ABISequence": 1
                    }
                  },
                  "Receiver": "eosmechanics",
                  "Act": {
                    "Account": "eosmechanics",
                    "Name": "cpu",
                    "Authorization": [
                      {
                        "actor": "eosmechanics",
                        "permission": "benchmark"
                      }
                    ],
                    "Data": "e30="
                  },
                  "ContextFree": false,
                  "Elapsed": 0,
                  "Console": "",
                  "AccountRamDeltas": [],
                  "Except": "",
                  "ErrorCode": 0,
                  "ReturnValue": ""
                }
              }
            ],
            "account_delta": null,
            "except": "",
            "error_code": 0,
            "failed_dtrx_trace": null,
            "partial": {
              "TypeID": 0,
              "Impl": {
                "Expiration": 1655725569,
                "RefBlockNum": 42586,
                "RefBlockPrefix": 1867795772,
                "MaxNetUsageWords": 0,
                "MaxCpuUsageMs": 0,
                "DelaySec": 0,
                "TransactionExtensions": [],
                "Signatures": [
                  "SIG_K1_K1C731F97cowJNpEmsYPwpyvvw5doMw46bNCMsyUKYaoF6jPfTn4KLy3XVqDLNY6k2HrU83J8spBf5E9kcU3TFTPXji5FT"
                ],
                "ContextFreeData": ""
              }
            }
          }
        }
      ]
    },
    "Deltas": null
  }
}
```

This issue is also not happening if we set `"IrreversibleOnly": true` in the block request options.

**Comment by heifner** [comment]
Are you running SHiP on a block producer node?

**Comment by matthewdarwin** [comment]
Nope, dedicated node.  3.1.0-rc1 build.

**Comment by heifner** [comment]
`head` is not updated until after `accepted_block` signal is emitted.

**Comment by heifner** [comment]
Resolved by #472
